main.go

```diff
@@ -167,7 +167,7 @@ func pay(ctx context.Context, hostIP, addressInput, amountInput, overridetokenIn
 	requestIPs[hostIP] = time.Now()
 	log.Infof("successfully sent %v to %v for %v",
 		amount, address, hostIP)
-	updateBalance(dcrwClient)
+	updateBalance(ctx, dcrwClient)
```

Comment:
if a user closes their connection between sendfrom and updateBalance, the balance won't be updated.

Perhaps we need a blockConnected() ntfn :)

Reply:
Good catch. I didn't know that a client closing the connection results in the context being cancelled.